e2e/app/page/notebook-page.ts

```diff
@@ -395,6 +397,42 @@ export default class NotebookPage extends NotebookFrame {
     return codeOutput;
   }
 
+  async runAllCells(): Promise<void> {
+    const clickCellMenuIcon = async (iframe: Frame): Promise<void> => {
+      await iframe.waitForXPath(Xpath.cellMenuDropdown, { visible: true, timeout: 2000 }).then((element) => {
+        element.hover();
+        element.click();
+      });
+    };
+
+    let maxRetries = 3;
```

Comment:
nit: because you are modifying this value, I would just call it retries instead. It represents the remaining retries , the initial value is the max